src_c/color.c

```diff
@@ -220,119 +220,54 @@ static PyGetSetDef _color_getsets[] = {
+    .sq_item = (ssizeargfunc)_color_item,
+    .sq_ass_item = (ssizeobjargproc)_color_ass_item,
 };
 
-static PyMappingMethods _color_as_mapping = {(lenfunc)_color_length,
-                                             (binaryfunc)_color_subscript,
-                                             (objobjargproc)_color_set_slice};
+static PyMappingMethods _color_as_mapping = {
+    .mp_length = (lenfunc)_color_length,
+    .mp_subscript = (binaryfunc)_color_subscript,
+    .mp_ass_subscript = (objobjargproc)_color_set_slice,
+};
 
 static PyBufferProcs _color_as_buffer = {(getbufferproc)_color_getbuffer,
                                          NULL};
 
-#define COLOR_TPFLAGS_COMMON \
-    (Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE | Py_TPFLAGS_CHECKTYPES)
-#define COLOR_TPFLAGS COLOR_TPFLAGS_COMMON
-
 #define DEFERRED_ADDRESS(ADDR) 0
 
 static PyTypeObject pgColor_Type = {
-    PyVarObject_HEAD_INIT(NULL, 0) "pygame.Color", /* tp_name */
-    sizeof(pgColorObject),                         /* tp_basicsize */
-    0,                                             /* tp_itemsize */
-    (destructor)_color_dealloc,                    /* tp_dealloc */
-    0,                                             /* tp_print */
-    NULL,                                          /* tp_getattr */
-    NULL,                                          /* tp_setattr */
-    NULL,                                          /* tp_compare */
-    (reprfunc)_color_repr,                         /* tp_repr */
-    &_color_as_number,                             /* tp_as_number */
-    &_color_as_sequence,                           /* tp_as_sequence */
-    &_color_as_mapping,                            /* tp_as_mapping */
-    NULL,                                          /* tp_hash */
-    NULL,                                          /* tp_call */
-    NULL,                                          /* tp_str */
-    NULL,                                          /* tp_getattro */
-    NULL,                                          /* tp_setattro */
-    &_color_as_buffer,                             /* tp_as_buffer */
-    COLOR_TPFLAGS,
-    DOC_PYGAMECOLOR,       /* tp_doc */
-    NULL,                  /* tp_traverse */
-    NULL,                  /* tp_clear */
-    _color_richcompare,    /* tp_richcompare */
-    0,                     /* tp_weaklistoffset */
-    NULL,                  /* tp_iter */
-    NULL,                  /* tp_iternext */
-    _color_methods,        /* tp_methods */
-    NULL,                  /* tp_members */
-    _color_getsets,        /* tp_getset */
-    NULL,                  /* tp_base */
-    NULL,                  /* tp_dict */
-    NULL,                  /* tp_descr_get */
-    NULL,                  /* tp_descr_set */
-    0,                     /* tp_dictoffset */
-    (initproc)_color_init, /* tp_init */
-    NULL,                  /* tp_alloc */
-    _color_new,            /* tp_new */
-#ifndef __SYMBIAN32__
```

Comment:
is id different for symbian? Or id this not used anymore?

Reply:
We probably don't actively support symbian ATM, atleast to my knowledge, so I think this change should be fine